src/create.rs

```diff
@@ -502,52 +502,37 @@ impl Party5 {
         Ok(Message5 { TX_f_signed_once })
     }
 
+    /// Returns the Channel and the transaction to broadcast.
     pub async fn receive(
         self,
         Message5 { TX_f_signed_once }: Message5,
         wallet: &impl SignFundingPSBT,
-    ) -> anyhow::Result<Party6> {
+    ) -> anyhow::Result<(Channel, Transaction)> {
```

Comment:
:question: Would it be overkill to return a struct with two fields instead?

Reply:
I'm allergic to returning tuples :sneezing_face:

Reply:
The reason I think a tuple is better is because the `Transaction` should be broadcasted and dropped straight away, meaning your 2 fields structs **must** be short lived.

By returning a tuple it communicates better to the consumer of this function: broadcast the `Transaction` and drop it, only keep the `Channel` instead of making them systematically deconstruct the returned value.